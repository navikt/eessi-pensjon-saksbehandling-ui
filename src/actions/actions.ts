export interface Action<T> {
  type: string,
  payload?: T
}

export interface JustTypeAction {
  type: string
}

export interface PayloadError {
  error: any
}
