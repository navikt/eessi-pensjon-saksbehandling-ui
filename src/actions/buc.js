
const BUCLIST = {

  ONGOING: [
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ']
    },
    {
      type: 'P_BUC_02',
      name: 'UførePensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['SE', 'DK', 'CZ'],
      merknader: ['foo', 'bar', 'baz']
    },
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ'],
      comments: ['foo', 'bar', 'baz']
    },
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ'],
      merknader: ['foo', 'bar', 'baz'],
      comments: ['foo', 'bar', 'baz']
    }
  ],
  OTHER: [
    {
      type: 'P_BUC_01',
      name: 'FIFOFA',
      dateCreated: 'dd.mm.åå',
      countries: ['RU', 'GB', 'CH']
    },
    {
      type: 'P_BUC_02',
      name: 'GRÅTASS',
      dateCreated: 'dd.mm.åå',
      countries: ['DE', 'TR', 'CA']
    }
  ]
}

const SEDS = [
  {
    name: 'P2000',
    status: 'sent',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P3000SE',
    status: 'draft',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P4000',
    status: 'received',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P5000',
    status: 'foo',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  }
]

export const fetchBucList = async () => {
  return BUCLIST
}

export const fetchSedListForBuc = async (buc) => {
  return SEDS
}
