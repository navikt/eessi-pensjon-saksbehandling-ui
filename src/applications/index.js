import SelectPDF from './PDF_/SelectPDF/SelectPDF'
import EditPDF from './PDF_/EditPDF/EditPDF'
import GeneratePDF from './PDF_/GeneratePDF/GeneratePDF'

import Case from './Case/Case'
import PSelv from './PSelv/PSelv'
import PInfo from './PInfo/PInfo'
import PInfoSaksbehandler from './PInfo/PInfoSaksbehandler'
import P4000 from './P4000/P4000'
import NewFeatures from './NewFeatures/NewFeatures'
import BUC from './BUC/BUC'

export {
  SelectPDF,
  EditPDF,
  GeneratePDF,
  Case,
  PSelv,
  PInfoSaksbehandler,
  PInfo,
  P4000,
  NewFeatures,
  BUC
}
