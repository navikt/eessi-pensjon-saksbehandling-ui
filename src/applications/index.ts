import BUC from './BUC'
import Journalføring from './Journalføring'
import PageNotification from './PageNotification'
import S3Inventory from './S3Inventory'

export {
  BUC,
  Journalføring,
  PageNotification,
  S3Inventory
}
