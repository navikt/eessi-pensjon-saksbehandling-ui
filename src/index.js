import 'babel-polyfill' // IE 11 compatibility
import 'core-js/es6/map' // IE 11 compatibility
import 'core-js/es6/set' // IE 11 compatibility
import 'es6-promise/auto' // IE 11 compatibility

import React, { Suspense } from 'react'
import ReactDOM from 'react-dom'
import createBrowserHistory from 'history/createBrowserHistory'
import { Switch, Redirect, Route, Router } from 'react-router'
import { createStore, applyMiddleware, combineReducers, compose } from 'redux'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'
import { I18nextProvider } from 'react-i18next'
import 'moment'
import 'moment/locale/en-gb'
import 'moment/locale/nb'

import i18n from './i18n'
import * as reducers from './reducers'
import * as routes from './constants/routes'
import { unregister } from './registerServiceWorker'
import * as Pages from './pages'
import AuthenticatedRoute from './components/app/AuthenticatedRoute'
import * as constants from './constants/constants'
import WaitingPanel from './components/app/WaitingPanel'

import 'bootstrap/dist/css/bootstrap.min.css'
import './index.css'
import './index_highContrast.css'

const history = createBrowserHistory()

const createStoreWithMiddleware = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore)

const reducer = combineReducers({
  ...reducers
})

const initialState = { ui: {
  language: i18n.language,
  locale: i18n.locale,
  modalOpen: false,
  modalBucketOpen: false,
  drawerEnabled: false,
  drawerOpen: false,
  footerOpen: false,
  drawerWidth: 10,
  drawerOldWidth: 250,
  highContrast: false
} }

const store = createStoreWithMiddleware(reducer, initialState)

ReactDOM.render(

  <I18nextProvider i18n={i18n}>
    <Provider store={store}>
      <Suspense fallback={<WaitingPanel/>}>
      <Router history={history}>
        <Switch>
          <AuthenticatedRoute exact path={routes.PSELV} component={Pages.PSelv} roles={[constants.SAKSBEHANDLER]} />
          <AuthenticatedRoute exact path={routes.PINFO} component={Pages.PInfo} roles={[constants.SAKSBEHANDLER, constants.BRUKER]} />
          <AuthenticatedRoute exact path={routes.PINFO_SAKSBEHANDLER} component={Pages.PInfoSaksbehandler} roles={[constants.SAKSBEHANDLER]} />
          <AuthenticatedRoute exact path={routes.P4000_ROUTE} component={Pages.P4000} roles={[constants.SAKSBEHANDLER, constants.BRUKER]} />
          <AuthenticatedRoute exact path={routes.PDF_GENERATE} component={Pages.GeneratePDF} roles={[constants.SAKSBEHANDLER, constants.BRUKER]} />
          <AuthenticatedRoute exact path={routes.PDF_EDIT} component={Pages.EditPDF} roles={[constants.SAKSBEHANDLER, constants.BRUKER]} />
          <AuthenticatedRoute exact path={routes.PDF_SELECT} component={Pages.SelectPDF} roles={[constants.SAKSBEHANDLER, constants.BRUKER]} />
          <Redirect from={routes.PDF} to={{ pathname: routes.PDF_SELECT }} />

          <AuthenticatedRoute exact path={routes.CASE_START} component={Pages.StartCase} roles={[constants.SAKSBEHANDLER]} />
          <AuthenticatedRoute exact path={routes.CASE_CONFIRM} component={Pages.ConfirmCase} roles={[constants.SAKSBEHANDLER]} />
          <AuthenticatedRoute exact path={routes.CASE_GENERATE} component={Pages.GenerateCase} roles={[constants.SAKSBEHANDLER]} />
          <AuthenticatedRoute exact path={routes.CASE_SAVE} component={Pages.SaveCase} roles={[constants.SAKSBEHANDLER]} />
          <AuthenticatedRoute exact path={routes.CASE_SEND} component={Pages.SendCase} roles={[constants.SAKSBEHANDLER]} />
          <Redirect from={routes.CASE} to={{ pathname: routes.CASE_START }} />

          <AuthenticatedRoute path={routes.INDEX} component={Pages.IndexPage} roles={[constants.SAKSBEHANDLER, constants.BRUKER]} />
          <Route path={routes.NOT_LOGGED} render={() => <Pages.Error type='notLogged' />} />
          <Route path={routes.NOT_INVITED} render={() => <Pages.Error type='notInvited' />} />
          <Route path={routes.FORBIDDEN} render={() => <Pages.Error type='forbidden' />} />
          <Route path={routes.ROOT + ':PATH+'} render={() => <Pages.Error type='error' />} />
          <AuthenticatedRoute path={routes.ROOT} component={Pages.FirstPage} roles={[constants.SAKSBEHANDLER, constants.BRUKER]} />
          <Redirect from='/' to={{ pathname: routes.ROOT, search: window.location.search }} />
        </Switch>
      </Router>
      </Suspense>
    </Provider>
  </I18nextProvider>,
  document.getElementById('root')
)

unregister()
// registerServiceWorker()
