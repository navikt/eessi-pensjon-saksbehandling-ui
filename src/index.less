@import 'nav-frontend-alertstriper-style/src/index';
@import 'nav-frontend-chevron-style/src/index';
@import 'nav-frontend-core/less/core';
@import 'nav-frontend-ekspanderbartpanel-style/src/index';
@import 'nav-frontend-etiketter-style/src/index';
@import 'nav-frontend-knapper-style/src/index';
@import 'nav-frontend-lenkepanel-style/src/index';
@import 'nav-frontend-lenker-style/src/lenker-style';
@import 'nav-frontend-lukknapp-style/src/lukknapp-style';
@import 'nav-frontend-modal-style/src/index';
@import 'nav-frontend-paneler-style/src/index';
@import 'nav-frontend-skjema-style/src/index';
@import 'nav-frontend-spinner-style/src/spinner-style';
@import 'nav-frontend-tabs-style/src/tabs-style';
@import 'nav-frontend-tabell-style/dist/main.css';
@import 'nav-frontend-typografi-style/src/index';
@import 'nav-frontend-typografi-style/src/fontimports';
@import 'nav-frontend-veileder-style/src/index';

html {
  height: 100%;
}

body {
  background-color: rgb(233, 231, 231);
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  height: 100vh;
  font-family: 'Source Sans Pro', Arial, sans-serif;
}

pre {
  font-family: 'Source Sans Pro', Arial, sans-serif;
}

code.block {
  display: block;
  white-space: pre-wrap
}

dd {
  margin-bottom: .5rem;
  margin-left: 0;
}

ol {
  list-style-type: decimal;
}

