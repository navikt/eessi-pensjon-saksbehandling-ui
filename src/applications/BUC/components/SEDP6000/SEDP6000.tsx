import { getSedP6000PDF, resetSedP6000PDF } from 'actions/buc'
import classNames from 'classnames'
import Modal from 'components/Modal/Modal'
import { AllowedLocaleString, ErrorElement } from 'declarations/app'
import { P6000 } from 'declarations/buc'
import { JoarkPreview } from 'declarations/joark'
import { State } from 'declarations/reducers'
import Flag from '@navikt/flagg-ikoner'
import File from '@navikt/forhandsvisningsfil'
import CountryData, { Country } from '@navikt/land-verktoy'
import _ from 'lodash'
import { Checkbox, Button, Loader, Panel } from '@navikt/ds-react'
import {
  FlexCenterDiv,
  FlexCenterSpacedDiv,
  HorizontalSeparatorDiv,
  PileDiv,
  VerticalSeparatorDiv
} from '@navikt/hoykontrast'
import { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'

export interface SEDP6000Props {
  feil: ErrorElement | undefined
  locale: AllowedLocaleString
  p6000s: Array<P6000>
  onChanged: (p6000s: Array<P6000>) => void
}

export interface SEDP6000Selector {
  gettingP6000PDF: boolean
  P6000PDF: JoarkPreview | null | undefined
}

const mapState = (state: State): SEDP6000Selector => ({
  gettingP6000PDF: state.loading.gettingP6000PDF,
  P6000PDF: state.buc.p6000PDF
})

const SEDP6000: React.FC<SEDP6000Props> = ({
  feil, locale, onChanged, p6000s
}: SEDP6000Props): JSX.Element => {
  const dispatch = useDispatch()
  const countryData = CountryData.getCountryInstance(locale)
  const { gettingP6000PDF, P6000PDF }: SEDP6000Selector = useSelector<State, SEDP6000Selector>(mapState)
  const { t } = useTranslation()
  const [chosenP6000s, setChosenP6000s] = useState<Array<P6000>>([])

  const changeChosenP6000 = (p6000: P6000, checked: boolean) => {
    let newChosenP6000s = _.cloneDeep(chosenP6000s)
    if (checked) {
      newChosenP6000s = newChosenP6000s.concat(p6000)
    } else {
      newChosenP6000s = _.filter(newChosenP6000s, _p6000 => _p6000.documentID !== p6000.documentID)
    }
    setChosenP6000s(newChosenP6000s)
    onChanged(newChosenP6000s)
  }

  const handlePreview = (bucId: string, docId: string) => {
    dispatch(getSedP6000PDF(bucId, docId))
  }

  const handleResetP6000 = () => {
    dispatch(resetSedP6000PDF())
  }

  return (
    <div id='a_buc_c_sedstart--p6000s-id'>
      <Modal
        open={!_.isNil(P6000PDF)}
        onModalClose={handleResetP6000}
        modal={{
          closeButton: true,
          modalContent: (
            <div style={{ cursor: 'pointer' }}>
              <File
                scale={2}
                file={{
                  size: P6000PDF?.filInnhold?.length ?? 0,
                  name: P6000PDF?.fileName ?? '',
                  mimetype: 'application/pdf',
                  content: {
                    base64: P6000PDF?.filInnhold
                  }
                }}
                width={1000}
                tema='simple'
                viewOnePage={false}
                onContentClick={handleResetP6000}
              />
            </div>
          )
        }}
      />
      {p6000s.map(p6000 => {
        const country: Country = countryData.findByValue(p6000.fraLand)
        return (
          <div className='a_buc_c_sedstart--p6000' key={p6000.documentID}>
            <Panel border className={classNames({ 'skjemaelement--input--harFeil': !!feil })}>
              <FlexCenterSpacedDiv>
                <FlexCenterDiv>
                  <Flag animate={false} wave={false} type='circle' country={p6000.fraLand} label={country.label ?? p6000.fraLand} />
                  <HorizontalSeparatorDiv />
                  <PileDiv>
                    <span>{t('ui:type')}: {p6000.type}</span>
                    <span>{t('ui:version')}: {p6000.sisteVersjon}</span>
                  </PileDiv>
                </FlexCenterDiv>
                <FlexCenterDiv>
                  <Button
                    variant='tertiary'
                    data-testid={'a_buc_c_sedstart--p6000-preview-' + p6000.documentID}
                    onClick={() => handlePreview(p6000.bucid, p6000.documentID)}
                    disabled={gettingP6000PDF}
                  >
                    {gettingP6000PDF && (
                      <>
                        <Loader />
                      </>
                    )}
                    {gettingP6000PDF ? t('ui:loading') : t('ui:preview')}
                  </Button>
                  <HorizontalSeparatorDiv />
                  <Checkbox
                    checked={_.find(chosenP6000s, _p6000 => _p6000.documentID === p6000.documentID) !== undefined}
                    key={p6000.documentID}
                    id={'a_buc_c_sedstart--p6000-checkbox-' + p6000.documentID}
                    data-testid={'a_buc_c_sedstart--p6000-checkbox-' + p6000.documentID}
                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => changeChosenP6000(p6000, e.target.checked)}

                  >{t('ui:choose')}
                  </Checkbox>
                </FlexCenterDiv>
              </FlexCenterSpacedDiv>
            </Panel>
            <VerticalSeparatorDiv />
          </div>
        )
      }
      )}
      {feil && (
        <div role='alert' aria-live='assertive' className='navds-error-message navds-error-message--medium navds-label'>
          {feil.feilmelding}
        </div>
      )}
    </div>
  )
}

export default SEDP6000
