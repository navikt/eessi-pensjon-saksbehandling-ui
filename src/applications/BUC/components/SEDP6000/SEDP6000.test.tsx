import { getSedP6000PDF, resetSedP6000PDF } from 'src/actions/buc'
import { render } from '@testing-library/react'
import { stageSelector } from 'src/setupTests'
import SEDP6000, { SEDP6000Props, SEDP6000Selector } from './SEDP6000'
import mockP6000s from 'src/mocks/buc/p6000'
import mockP6000PDF from 'src/mocks/buc/p6000pdf'

const defaultSelector: SEDP6000Selector = {
  gettingP6000PDF: false,
  P6000PDF: undefined
}

jest.mock('actions/buc', () => ({
  getSedP6000PDF: jest.fn(),
  resetSedP6000PDF: jest.fn()
}))

jest.mock('components/Modal/Modal', () => (props: any) => (
  <div className='mock-c-modal' tabIndex={0} role='button' onKeyPress={props.onModalClose} onClick={props.onModalClose}>{props.children}</div>
))

describe('applications/BUC/components/SEDP6000/SEDP6000', () => {
  let wrapper: any
  const initialMockProps: SEDP6000Props = {
    feil: undefined,
    locale: 'nb',
    p6000s: mockP6000s,
    onChanged: jest.fn()
  }

  beforeEach(() => {
    stageSelector(defaultSelector, {})
    wrapper = render(<SEDP6000 {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    const { container } = render(<SEDP6000 {...initialMockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    expect(wrapper.find('div.a_buc_c_sedstart--p6000').hostNodes().length).toEqual(mockP6000s.length)
  })

  it('Handling: preview click', () => {
    (getSedP6000PDF as jest.Mock).mockReset()
    wrapper.find('[data-testid=\'a_buc_c_sedstart--p6000-preview-' + mockP6000s[0].documentID + '\']')
      .hostNodes().simulate('click')
    expect(getSedP6000PDF).toHaveBeenCalled()
  })

  it('Handling: select click', () => {
    (initialMockProps.onChanged as jest.Mock).mockReset()
    wrapper.find('[data-testid=\'a_buc_c_sedstart--p6000-checkbox-' + mockP6000s[0].documentID + '\']')
      .hostNodes().simulate('change', { target: { checked: true } })
    expect(initialMockProps.onChanged).toHaveBeenCalled()
  })

  it('Handling: modal close', () => {
    (resetSedP6000PDF as jest.Mock).mockReset()
    stageSelector(defaultSelector, { P6000PDF: mockP6000PDF })
    wrapper = render(<SEDP6000 {...initialMockProps} />)
    wrapper.find('div.mock-c-modal').simulate('click')
    expect(resetSedP6000PDF).toHaveBeenCalled()
  })
})
