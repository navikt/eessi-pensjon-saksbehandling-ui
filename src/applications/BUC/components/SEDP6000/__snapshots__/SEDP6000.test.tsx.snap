// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDP6000/SEDP6000 Render: match snapshot 1`] = `
.c9 {
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
  display: inline-block;
  line-height: 0px;
  margin: 0.25rem;
}

.c9.animate {
  opacity: 0.01;
  -webkit-animation: iPdXhn 0.5s forwards;
  animation: iPdXhn 0.5s forwards;
}

.c9.type-original.wave div {
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-animation: eRATHR 4s ease-in-out infinite;
  animation: eRATHR 4s ease-in-out infinite;
}

.c9.type-circle.wave div img {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-animation: eRATHR 4s ease-in-out infinite;
  animation: eRATHR 4s ease-in-out infinite;
}

ui-fss.g1[id="c8"] {
  content: "c9,";
}

.c11.type-circle {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 1px solid var(--navds-semantic-color-border);
  border-radius: 50%;
  overflow: hidden;
  position: relative;
}

.c11.type-circle img {
  position: absolute;
  margin: auto;
  top: -999px;
  left: -999px;
  bottom: -999px;
  right: -999px;
}

.c11.type-circle.size-XS {
  width: 16px;
  height: 16px;
  min-width: 16px;
  min-height: 16px;
  max-width: 16px;
  max-height: 16px;
}

.c11.type-circle.size-XS img {
  height: 18px;
}

.c11.type-circle.size-S {
  width: 24px;
  height: 24px;
  min-width: 24px;
  min-height: 24px;
  max-width: 24px;
  max-height: 24px;
}

.c11.type-circle.size-S img {
  height: 26px;
}

.c11.type-circle.size-M {
  width: 32px;
  height: 32px;
  min-width: 32px;
  min-height: 32px;
  max-width: 32px;
  max-height: 32px;
}

.c11.type-circle.size-M img {
  height: 35px;
}

.c11.type-circle.size-L {
  width: 40px;
  height: 40px;
  min-width: 40px;
  min-height: 40px;
  max-width: 40px;
  max-height: 40px;
}

.c11.type-circle.size-L img {
  height: 45px;
}

.c11.type-circle.size-XL {
  width: 48px;
  height: 48px;
  min-width: 48px;
  min-height: 48px;
  max-width: 48px;
  max-height: 48px;
}

.c11.type-circle.size-XL img {
  height: 54px;
}

.c11.type-circle.size-XXL {
  width: 72px;
  height: 72px;
  min-width: 72px;
  min-height: 72px;
  max-width: 72px;
  max-height: 72px;
}

.c11.type-circle.size-XXL img {
  height: 82px;
}

.c11.type-original.size-XS img {
  height: 16px;
  min-height: 16px;
  width: auto;
}

.c11.type-original.size-S img {
  height: 24px;
  min-height: 24px;
  width: auto;
}

.c11.type-original.size-M img {
  height: 32px;
  min-height: 32px;
  width: auto;
}

.c11.type-original.size-L img {
  height: 40px;
  min-height: 40px;
  width: auto;
}

.c11.type-original.size-XL img {
  height: 48px;
  min-height: 48px;
  width: auto;
}

.c11.type-original.size-XXL img {
  height: 72px;
  min-height: 72px;
  width: auto;
}

ui-fss.g2[id="c10"] {
  content: "c11,";
}

.c7 {
  padding: 0.25rem;
}

ui-fss.g5[id="c6"] {
  content: "c7,";
}

.c13 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g23[id="c12"] {
  content: "c13,";
}

.c17 {
  margin-bottom: 1rem;
}

ui-fss.g24[id="c16"] {
  content: "c17,";
}

.c3 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

ui-fss.g25[id="c0"] {
  content: "c3,";
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

ui-fss.g28[id="c1"] {
  content: "c4,";
}

.c5 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

ui-fss.g32[id="c2"] {
  content: "c5,";
}

.c15 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g36[id="c14"] {
  content: "c15,";
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c11.type-original {
    display: inline-block;
  }

  .c11.type-original img {
    border: 1px solid var(--navds-semantic-color-border);
  }
}

@supports not (-ms-high-contrast:none) {
  .c11.type-original {
    border: 1px solid var(--navds-semantic-color-border);
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
  }
}

<div
  id="a_buc_c_sedstart--p6000s-id"
>
  <div
    class="mock-c-modal"
    role="button"
    tabindex="0"
  />
  <div
    class="a_buc_c_sedstart--p6000"
  >
    <div
      class="navds-panel navds-panel--border"
    >
      <div
        class="c0 c1 c2 c3 c4 c5"
      >
        <div
          class="c0 c1 c3 c4"
        >
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="top"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <span
              class="c6 c7"
            >
              Sverige
            </span>
            <div
              class="navds-popover__arrow"
              style="bottom: -0.5rem;"
            />
          </div>
          <div
            id="flag-ab927ebe-c68a-457b-96a9-248a8a1baeb8"
            style="display: inline-block;"
          >
            <div
              class="c8 c9 size-M type-circle"
            >
              <div
                class="c10 c11 size-M type-circle"
              >
                <img
                  alt="Sverige"
                  src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 10'%3e %3crect width='16' height='10' fill='%23006aa7'/%3e %3crect width='2' height='10' x='5' fill='%23fecc00'/%3e %3crect width='16' height='2' y='4' fill='%23fecc00'/%3e %3c/svg%3e"
                />
              </div>
            </div>
          </div>
          <div
            class="c12 c13"
          />
          <div
            class="c14 c15"
          >
            <span>
              ui:type
              : 
              p6000
            </span>
            <span>
              ui:version
              : 
              1
            </span>
          </div>
        </div>
        <div
          class="c0 c1 c3 c4"
        >
          <button
            class="navds-button navds-button--tertiary navds-button--medium"
            data-testid="a_buc_c_sedstart--p6000-preview-123123123"
          >
            <span
              aria-live="polite"
              class="navds-label"
            >
              ui:preview
            </span>
          </button>
          <div
            class="c12 c13"
          />
          <div
            class="navds-checkbox navds-checkbox--medium"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="navds-checkbox__input"
              data-testid="a_buc_c_sedstart--p6000-checkbox-123123123"
              id="a_buc_c_sedstart--p6000-checkbox-123123123"
              type="checkbox"
            />
            <label
              class="navds-checkbox__label"
              for="a_buc_c_sedstart--p6000-checkbox-123123123"
            >
              <span
                class="navds-checkbox__content"
              >
                <span
                  class="navds-body-short"
                >
                  ui:choose
                </span>
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c16 c17"
    />
  </div>
  <div
    class="a_buc_c_sedstart--p6000"
  >
    <div
      class="navds-panel navds-panel--border"
    >
      <div
        class="c0 c1 c2 c3 c4 c5"
      >
        <div
          class="c0 c1 c3 c4"
        >
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="top"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <span
              class="c6 c7"
            >
              Danmark
            </span>
            <div
              class="navds-popover__arrow"
              style="bottom: -0.5rem;"
            />
          </div>
          <div
            id="flag-0bcdddf2-4477-4150-8862-8a155ac52b67"
            style="display: inline-block;"
          >
            <div
              class="c8 c9 size-M type-circle"
            >
              <div
                class="c10 c11 size-M type-circle"
              >
                <img
                  alt="Danmark"
                  src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 370 280'%3e %3crect width='370' height='280' fill='%23c60c30'/%3e %3crect width='40' height='280' x='120' fill='%23fff'/%3e %3crect width='370' height='40' y='120' fill='%23fff'/%3e %3c/svg%3e"
                />
              </div>
            </div>
          </div>
          <div
            class="c12 c13"
          />
          <div
            class="c14 c15"
          >
            <span>
              ui:type
              : 
              p6000
            </span>
            <span>
              ui:version
              : 
              2
            </span>
          </div>
        </div>
        <div
          class="c0 c1 c3 c4"
        >
          <button
            class="navds-button navds-button--tertiary navds-button--medium"
            data-testid="a_buc_c_sedstart--p6000-preview-123123124"
          >
            <span
              aria-live="polite"
              class="navds-label"
            >
              ui:preview
            </span>
          </button>
          <div
            class="c12 c13"
          />
          <div
            class="navds-checkbox navds-checkbox--medium"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="navds-checkbox__input"
              data-testid="a_buc_c_sedstart--p6000-checkbox-123123124"
              id="a_buc_c_sedstart--p6000-checkbox-123123124"
              type="checkbox"
            />
            <label
              class="navds-checkbox__label"
              for="a_buc_c_sedstart--p6000-checkbox-123123124"
            >
              <span
                class="navds-checkbox__content"
              >
                <span
                  class="navds-body-short"
                >
                  ui:choose
                </span>
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c16 c17"
    />
  </div>
  <div
    class="a_buc_c_sedstart--p6000"
  >
    <div
      class="navds-panel navds-panel--border"
    >
      <div
        class="c0 c1 c2 c3 c4 c5"
      >
        <div
          class="c0 c1 c3 c4"
        >
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="top"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <span
              class="c6 c7"
            >
              Finland
            </span>
            <div
              class="navds-popover__arrow"
              style="bottom: -0.5rem;"
            />
          </div>
          <div
            id="flag-d20c589c-0e4e-4150-8606-2d71f7b800d8"
            style="display: inline-block;"
          >
            <div
              class="c8 c9 size-M type-circle"
            >
              <div
                class="c10 c11 size-M type-circle"
              >
                <img
                  alt="Finland"
                  src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1800 1100'%3e %3crect width='1800' height='1100' fill='%23fff'/%3e %3crect width='1800' height='300' y='400' fill='%23003580'/%3e %3crect width='300' height='1100' x='500' fill='%23003580'/%3e %3c/svg%3e"
                />
              </div>
            </div>
          </div>
          <div
            class="c12 c13"
          />
          <div
            class="c14 c15"
          >
            <span>
              ui:type
              : 
              p6000
            </span>
            <span>
              ui:version
              : 
              3
            </span>
          </div>
        </div>
        <div
          class="c0 c1 c3 c4"
        >
          <button
            class="navds-button navds-button--tertiary navds-button--medium"
            data-testid="a_buc_c_sedstart--p6000-preview-123123125"
          >
            <span
              aria-live="polite"
              class="navds-label"
            >
              ui:preview
            </span>
          </button>
          <div
            class="c12 c13"
          />
          <div
            class="navds-checkbox navds-checkbox--medium"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="navds-checkbox__input"
              data-testid="a_buc_c_sedstart--p6000-checkbox-123123125"
              id="a_buc_c_sedstart--p6000-checkbox-123123125"
              type="checkbox"
            />
            <label
              class="navds-checkbox__label"
              for="a_buc_c_sedstart--p6000-checkbox-123123125"
            >
              <span
                class="navds-checkbox__content"
              >
                <span
                  class="navds-body-short"
                >
                  ui:choose
                </span>
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c16 c17"
    />
  </div>
</div>
`;
